### DISCUSSION TITLE
Charges with background/foreground

### DISCUSSION THREAD

**Discussion by Blipz** [Azgaar/Armoria#168] [Ideas]
For some charges, it would be useful to define some of their parts as background, in order to allow things like people/animals holding objects, or supporters around the shield.

Proposed implementation:
- add a "background" class to parts of the SVG
- add a boolean field in the data model for declaring charges that support the feature
- at the end of the charge rendering, redraw the relevant charges (if the user checks the corresponding option) but with the background parts hidden

This might not be trivial for some charges and require to redraw some SVG paths. However, the number of charges affected is quite limited.

**Comment by Azgaar** [comment]
Interesting idea. Svg supports holes, so it also can be just not element there so it will show what's behind. But sometimes it's too hard to do it in svg.

**Comment by Blipz** [comment]
I don't see how it would work with holes, the holes need to depend on the shape and position of other charges.

**Reply by Azgaar** [reply]
Yes, sure

**Comment by Blipz** [comment]
Done here: https://github.com/Azgaar/Armoria/pull/172
